packages/amplify_datastore/example/integration_test/relationship_type_test.dart

```diff
@@ -0,0 +1,366 @@
+        expect(children, isEmpty);
+      });
+    });
+
+    group('BelongsTo', () {
+      var child = ChildModel(name: 'child');
+      var parent = BelongsToModel(name: 'HasOne', child: child);
+      late Future<SubscriptionEvent<ChildModel>> belongsToEvent;
+      late Future<SubscriptionEvent<BelongsToModel>> hasOneEvent;
+
+      setUpAll(() async {
+        await configureDataStore();
+        await clearDataStore();
+
+        belongsToEvent = Amplify.DataStore.observe(ChildModel.classType).first;
+        hasOneEvent = Amplify.DataStore.observe(BelongsToModel.classType).first;
+      });
+
+      testWidgets('precondition', (WidgetTester tester) async {
+        var queriedChildren =
+            await Amplify.DataStore.query(ChildModel.classType);
+        expect(queriedChildren, isEmpty);
+        var queriedParents =
+            await Amplify.DataStore.query(BelongsToModel.classType);
+        expect(queriedParents, isEmpty);
+      });
+
+      testWidgets('save child', (WidgetTester tester) async {
+        await Amplify.DataStore.save(child);
+        var children = await Amplify.DataStore.query(ChildModel.classType);
+        expect(children, isNotEmpty);
+      });
+
+      testWidgets('save parent', (WidgetTester tester) async {
+        await Amplify.DataStore.save(parent);
+        var parents = await Amplify.DataStore.query(BelongsToModel.classType);
+        expect(parents, isNotEmpty);
+      });
+
+      testWidgets('query parent', (WidgetTester tester) async {
+        var parents = await Amplify.DataStore.query(BelongsToModel.classType);
+        var queriedParent = parents.first;
+        expect(queriedParent, parent);
+        expect(queriedParent.child, child);
+      });
+
+      testWidgets('query child', (WidgetTester tester) async {
+        var children = await Amplify.DataStore.query(ChildModel.classType);
+        var queriedChild = children.first;
+        expect(queriedChild, child);
+      });
+
+      testWidgets('observe parent', (WidgetTester tester) async {
+        var event = await hasOneEvent;
+        var observedParent = event.item;
+        expect(observedParent, parent);
+        expect(observedParent.child, child);
+      });
+
+      testWidgets('observe child', (WidgetTester tester) async {
+        var event = await belongsToEvent;
+        var observedChild = event.item;
+        expect(observedChild, child);
+      });
+
+      testWidgets('delete parent', (WidgetTester tester) async {
+        await Amplify.DataStore.delete(parent);
+        var parents = await Amplify.DataStore.query(BelongsToModel.classType);
+        expect(parents, isEmpty);
+      });
+
+      testWidgets('delete child', (WidgetTester tester) async {
+        await Amplify.DataStore.delete(child);
+        var children = await Amplify.DataStore.query(ChildModel.classType);
+        expect(children, isEmpty);
+      });
+    });
+
+    group('hasMany', () {
```

Comment:
Nit for consistency:
```suggestion
    group('HasMany', () {
```